plugins/session_notifier.rb

```diff
@@ -237,6 +270,34 @@ def send_text_to_dingtalk(session)
         print_status((body['errcode'] == 0) ? 'Session notified to DingTalk.' : 'Failed to send notification.')
       end
 
+      def send_text_to_gotify(session)
+        # https://gotify.net/docs/more-pushmsg
+        uri_parser = URI.parse(gotify_address)
+        message_text =
+        "Platform : #{session.platform}\n" \
+        "Tunnel : #{session.tunnel_to_s}\n" \
+        "Arch : #{session.arch}\n" \
+        "Info : > #{session.info ? session.info.to_s : nil}"
+        json_post_data = JSON.pretty_generate({
+          title: "A #{session.platform}/#{session.type} Session is On!",
+          message: message_text,
+          priority: 10
+        })
+        http = Net::HTTP.new(uri_parser.host, uri_parser.port)
+        if !gotify_sslcert.nil?
+          http.use_ssl = true
+          http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+          store = OpenSSL::X509::Store.new
+          store.add_file(gotify_sslcert)
+        end
+        request = Net::HTTP::Post.new(uri_parser.request_uri)
+        request.content_type = 'application/json'
+        request.body = json_post_data
+        res = http.request(request)
+        body = JSON.parse(res.body)
+        print_status((body['priority'] == 10) ? 'Session notified to Gotify.' : 'Failed to send notification.')
```

Comment:
Techinically speaking this and the code above should assume that the HTTP request may not complete successfully and therefore `JSON.parse` may not return a valid object. Therefore there is no guarantee that `body['priority']` will actually exist.

Reply:
Part about the return value from `JSON.parse` wasn't true but I did add in some extra checks to fix the remaining issues discussed here.